File: src/main/java/leets/leenk/domain/media/application/usecase/MediaUsecase.java
package leets.leenk.domain.media.application.usecase;

import leets.leenk.domain.media.application.dto.response.MediaUrlResponse;
import leets.leenk.domain.media.domain.entity.Media;
import leets.leenk.domain.media.domain.entity.enums.ContentType;
import leets.leenk.domain.media.domain.service.MediaGetService;
import leets.leenk.domain.media.domain.service.MediaUpdateService;
import leets.leenk.domain.media.domain.service.S3PresignedUrlService;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@RequiredArgsConstructor
public class MediaUsecase {

    private final S3PresignedUrlService s3PresignedUrlService;
    private final MediaGetService mediaGetService;
    private final MediaUpdateService mediaUpdateService;

    public List<MediaUrlResponse> getUrl(ContentType contentType, List<String> fileName) {

        return s3PresignedUrlService.generateUrlList(contentType, fileName);
    }

    @Transactional
    public void updateThumbnailUrl(String originalUrl, String thumbnailUrl){
        Media media = mediaGetService.findByMediaUrl(originalUrl);

        mediaUpdateService.update(media, thumbnailUrl);
    }
}

File: src/main/kotlin/leets/leenk/domain/media/application/usecase/MediaUsecase.kt
package leets.leenk.domain.media.application.usecase

import leets.leenk.domain.media.application.dto.response.MediaUrlResponse
import leets.leenk.domain.media.domain.entity.enums.DomainType
import leets.leenk.domain.media.domain.service.MediaGetService
import leets.leenk.domain.media.domain.service.MediaUpdateService
import leets.leenk.domain.media.domain.service.S3PresignedUrlService
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class MediaUsecase(
    private val s3PresignedUrlService: S3PresignedUrlService,
    private val mediaGetService: MediaGetService,
    private val mediaUpdateService: MediaUpdateService,
) {
    fun getUrl(
        domainType: DomainType,
        fileName: List<String>,
    ): List<MediaUrlResponse> = s3PresignedUrlService.generateUrlList(domainType, fileName)

    @Transactional
    fun updateThumbnailUrl(
        originalUrl: String,
        thumbnailUrl: String,
    ) {
        val media = mediaGetService.findByMediaUrl(originalUrl)
        mediaUpdateService.update(media, thumbnailUrl)
    }
}
